File: LeadToOffer/15_ReverseList.cpp
// O(1) time.
class Solution {
public:
    ListNode* ReverseList(ListNode* pHead) {
        if (!pHead)
            return NULL;

        ListNode* pFirst = pHead;
        ListNode* temp = NULL;
        while (pHead->next) {
            temp = pHead->next;
            pHead->next = temp->next;
            temp->next = pFirst;
            pFirst = temp;
        }
        return pFirst;
    }
};

File: LeadToOffer/15_ReverseList.py
## O(n) time.
class Solution:
    def ReverseList(self, pHead):
        if pHead is None:
            return None

        pFirst = pHead
        while pHead.next is not None:
            temp = pHead.next
            pHead.next = temp.next
            temp.next = pFirst
            pFirst = temp
        return pFirst
